internal/context.go

```diff
@@ -185,6 +185,12 @@ func WithCancel(parent Context) (ctx Context, cancel CancelFunc) {
 	return c, func() { c.cancel(true, ErrCanceled) }
 }
 
+// NewDisconnectedContext returns a new context that won't propagate parent's cancellation to the new child context.
```

Comment:
Would you add the following sample:

```
       err := workflow.ExecuteActivity(ctx, activitity1).Get(ctx, &activity1Result)
	if err != nil && cadence.IsCanceledError(err) {
                disconnectedCtx, _ := workflow.newDisconnectedContext(ctx);
		workflow.ExecuteActivity(disconnectedCtx, handleCancellation)
		return err
	}
```

Reply:
I think in this particular case it is equivalent, but uglier.

Reply:
There is slightly different. Activity could voluntarily return CanceledError. In that case, workflow context is not canceled.

Reply:
Also the workflow needs to wait until the handleCancellation activity returns, otherwise workflow returns CanceledError that would close the workflow and no activity would be scheduled..